app/src/main/java/de/danoeh/antennapod/fragment/CompletedDownloadsFragment.java

```diff
@@ -280,6 +319,11 @@ private void loadItems() {
         emptyView.hide();
         disposable = Observable.fromCallable(() -> {
             List<FeedItem> downloadedItems = DBReader.getDownloadedItems();
+
+            SortOrder sortOrder = getDownloadsSortedOrder();
+            final Permutor<FeedItem> permutor = FeedItemPermutors.getPermutor(sortOrder);
+            permutor.reorder(downloadedItems);
```

Comment:
My idea would be to let the database do the sorting, similar to what is done in https://github.com/AntennaPod/AntennaPod/pull/6266/files. That way, we can have lazy loading. Also, it's much faster. Could you please extend the `FeedItemSortQuery.java` class by @gitstart and add it to your PR? That class can then also be reused in the other PR.

Reply:
@ByteHamster Sure, no problem. I am almost done implementing the changes. However, I have a small question: Do you know what the database key for the podcast title in the `PodDBAdapter` is? AFAIK the `KEY_TITLE` key only stores the titles of the individual episodes. If you don't know right away, that's no problem. I will simply have to try several keys until I find the right one.

Reply:
Uhm, right. I think the same key is used for both titles. At least currently it does not seem to be a problem because we never join the `FeedItems` and `Feeds` tables. But when selecting the sort order "by feed title", we do need to join them. Maybe just ignore that sort order for now. I still need to think about how to best solve this.

Reply:
Ok, I will do that. Thanks for the quick reply.